### PR TITLE
Rearrange SC validation in prep for compute shaders

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1635,7 +1614,7 @@ static void collect_interface_by_descriptor_slot(layer_data *my_data, VkDevice d
             auto existing_it = out.find(std::make_pair(set, binding));
             if (existing_it != out.end()) {
                 /* conflict within spv image */
-                log_msg(my_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT, /*dev*/ 0,
+                log_msg(my_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT, 0,
```
Comment (tobine) [inline layers/core_validation.cpp]: Do we want to change the "VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT" param as well?

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -2490,7 +2469,7 @@ static bool descriptor_type_match(layer_data *my_data, shader_module const *modu
 static VkBool32 require_feature(layer_data *my_data, VkBool32 feature, char const *feature_name) {
     if (!feature) {
         if (log_msg(my_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT,
```
Comment (tobine) [inline layers/core_validation.cpp]: Same comment here, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT no longer valid.

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -2642,12 +2621,102 @@ static VkBool32 validate_shader_capabilities(layer_data *my_data, VkDevice dev,
...
+                                               spirv_inst_iter *out_entrypoint)
+{
+    VkBool32 pass = VK_TRUE;
+    auto module = *out_module = dev_data->shaderModuleMap[pStage->module].get();
+    pass &= validate_specialization_offsets(dev_data, pStage);
+
+    /* find the entrypoint */
+    auto entrypoint = *out_entrypoint = find_entrypoint(module, pStage->pName, pStage->stage);
+    if (entrypoint == module->end()) {
+        if (log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT,
```
Comment (tobine) [inline layers/core_validation.cpp]: same, object type enum VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT now invalid

Comment (chrisforbes) [inline layers/core_validation.cpp]: Well, it never was... no behavior change here. I've added a commit to stop claiming device 0 is interesting though.

**Comment by tobine** [comment]
:+1: merged to master

**Pull request closed without merging**